lib/class-wp-theme-json-gutenberg.php

```diff
@@ -387,6 +404,13 @@ private static function get_blocks_metadata() {
 				self::$blocks_metadata[ $block_name ]['selector'] = '.wp-block-' . str_replace( '/', '-', str_replace( 'core/', '', $block_name ) );
 			}
 
+			if (
+				isset( $block_type->supports['color']['__experimentalDuotone'] ) &&
+				is_string( $block_type->supports['color']['__experimentalDuotone'] )
+			) {
+				self::$blocks_metadata[ $block_name ]['filter'] = $block_type->supports['color']['__experimentalDuotone'];
```

Comment:
The idea is that we register the selector for the property using the property name (`filter`) instead of an intermediary key (`duotone`). This way, we compute all the declarations for the node and only then decide which ones use the general selector and which ones use the property selector.

Reply:
The problem that I was trying to solve by having a separate `DUOTONE_PROPERTIES_METADATA` was to account for wanting different selectors for different filters. Let's say, for the media and text block, we want to add duotone to the media and a drop shadow to the text. Duotone and drop shadow both use the `filter` CSS property, so it's not a simple one-to-one mapping anymore. All duotones are filters, but not all filters are duotone.

Reply:
I can't seem to get the code working. What is the structure I should be using for theme.json with this change?

Reply:
> I can't seem to get the code working. What is the structure I should be using for theme.json with this change?

Nothing, really, it works the same. It's just an internal change.

<details>
<summary>
For reference, this is the theme.json I am using (with and without the change):
</summary>

```json
{
  "version": 1,
  "settings": {
    "color": {
      "duotone": [],
      "palette": []
    }
  },
  "styles": {
    "blocks": {
      "core/cover": {
        "color": {
          "duotone": "var(--wp--preset--duotone-filter--magenta-yellow)"
        }
      },
      "core/image": {
        "color": {
          "duotone": "var(--wp--preset--duotone-filter--magenta-yellow)"
        }
      },
      "core/post-author": {
        "color": {
          "duotone": "var(--wp--preset--duotone-filter--magenta-yellow)"
        }
      },
      "core/post-featured-image": {
        "color": {
          "duotone": "var(--wp--preset--duotone-filter--magenta-yellow)"
        }
      }
    }
  }
}

```
</details>

> The problem that I was trying to solve by having a separate DUOTONE_PROPERTIES_METADATA was to account for wanting different selectors for different filters.

Nice. I came to this from the perspective of any other property that wants to do the same. I was concerned with having a single meta array for any property that wants its selector.

Said this, we can land this PR without having to solve that issue yet. There's no other property (filter or not) that needs this at the moment. I'd suggest crossing that bridge when we get to it. With or without https://github.com/WordPress/gutenberg/pull/34667/commits/3c2c1666793f85b5da30f1424b2e4467a6b6727c this works for me, so feel free to revert. Note that we need to make sure that `remove_insecure_styles` sanitizes all the properties that we have.

Reply:
I've gone ahead and reverted my change at https://github.com/WordPress/gutenberg/pull/34667/commits/220d86d64a9483a5b717e02ff9d225c3c4856510